src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -1383,8 +1572,13 @@ private function parse_next_tag() {
 					// Abruptly-closed empty comments are a sequence of dashes followed by `>`.
```

Comment:
It's a bit outside this diff, but I'm confused by somme of the comments just before this section:

https://github.com/WordPress/wordpress-develop/blob/103a556b6095fe61843719933db7ce2d0688ec7a/src/wp-includes/html-api/class-wp-html-tag-processor.php#L1550-L1562

`<!` transitions to "markup declaration open state", then the following `--` switch to "comment start state". So all of `<!--` should move us into "comment start state" not "bogus comment state."

Reply:
rearranged comments in 9f2992070ebbbb8970c96c115da2fb74a7ebfda4. there has been significant shuffle to this part of the code and I think the comment had become lost at some point.